fabtests/common/shared.c

```diff
@@ -2770,17 +2802,36 @@ static int ft_fdwait_for_comp(struct fid_cq *cq, uint64_t *cur,
 {
 	struct fi_cq_err_entry comp;
 	struct fid *fids[1];
-	int fd, ret;
+	int fd, ret = 0;
+	int remaining = timeout;
 
 	fd = cq == txcq ? tx_fd : rx_fd;
 	fids[0] = &cq->fid;
 
 	while (total - *cur > 0) {
 		ret = fi_trywait(fabric, fids, 1);
 		if (ret == FI_SUCCESS) {
-			ret = ft_poll_fd(fd, timeout);
+			int wait_timeout = ft_fdwait_need_progress
+				? ft_fdwait_poll_timeout(remaining)
+				: remaining;
```

Comment:
the same condition is checked inside `ft_fdwait_poll_timeout()`.  can eliminate this variable and use `ft_fdwait_poll_timeout(remaining)` directly.